### DISCUSSION TITLE
Is it possible to write the declaration section in python

### DISCUSSION THREAD

**Discussion by fkgl** [UPPAALModelChecker/UPPAAL-Meta#184] [Q&A] [answered]
Hello everybody,

Please, is it possible to write the **declaration section** in **python** (not in Java as shown in the examples)?
If not, is there a way to have the xml file contain _only_ the templates and call the declarations (functions, constants, ... etc) from the **python project**?

**Comment by thorulf4** [comment, marked as answer]
Unfortunately no, Uppaal does not support other languages than the default language. Which is not java but a custom variant of C. More information about this language is available [here](https://docs.uppaal.org/language-reference/system-description/declarations/).

The closest we can get is calling external C libraries, these libraries could in theory interop with python. But I wouldn't recommend using  this feature, as [external functions](https://docs.uppaal.org/language-reference/system-description/declarations/external-functions/) doesn't actually solve your issue.

**Reply by fkgl** [reply]
Thanks so much for your reply
